### PR TITLE
Flesh out docs + doctests

### PR DESCRIPTION
**Pull request by logannc**
Flesh out docs on all public methods and move some tests into doctests where appropriate.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,5 +1,10 @@
 # fuzzywuzzy-rs [![docs.rs badge](https://docs.rs/fuzzywuzzy/badge.svg)](https://docs.rs/fuzzywuzzy) [![crates.io badge](https://img.shields.io/crates/v/fuzzywuzzy.svg)](https://crates.io/crates/fuzzywuzzy)
```
Comment (seanpianka) [inline README.md]: Perhaps this should go onto the line below? I personally prefer the look on separate lines.

File: README.md [hunk 2]
```diff
@@ -1,5 +1,10 @@
...
-Fuzzy string matching like a boss. It uses Levenshtein Distance to calculate the differences between sequences in a simple-to-use package.
+> Fuzzy string matching like a boss. It uses Levenshtein Distance to calculate the differences between sequences in a simple-to-use package.
+> [fuzzywuzzy](https://github.com/seatgeek/fuzzywuzzy)
+
+This is a Rust clone of the python package fuzzywuzzy. We attempt to be 100% compatible with the python version.
```
Comment (seanpianka) [inline README.md]:
```suggestion
This is a Rust port of the Python package fuzzywuzzy. We aim to be drop-in replacement for the original.
```

File: README.md (hunk 1 above)
Comment (logannc) [reply README.md]: Done.

File: README.md [hunk 3]
```diff
@@ -1,5 +1,10 @@
...
+
+At the time of writing, our matching algorithm is based on the difflib implementation results which may, in rare cases, [have slightly different results](https://github.com/seatgeek/fuzzywuzzy/issues/128) compared to the python Levenshtein implementation.
```
Comment (seanpianka) [inline README.md]:
```suggestion
However, our matching algorithm is based on [difflib](https://github.com/DimaKudosh/difflib/wiki), rather than original's Levenshtein-based implementation. This may, in rare cases, cause the ported functions to [behave differently than their original counterpart](https://github.com/seatgeek/fuzzywuzzy/issues/128).
```

File: src/fuzz.rs [hunk 4]
```diff
@@ -265,8 +278,9 @@ mod tests {
         s9a: &'static str,
         s10: &'static str,
         s10a: &'static str,
-        // TODO: Test silly corner cases,
+        #[allow(dead_code)]
         cirque_strings: &'static [&'static str; 6],
```
Comment (seanpianka) [inline src/fuzz.rs]: I added these blindly while porting the unit tests, however neither the original nor our ported tests make use of these. Shall we delete them?

Comment (logannc) [reply src/fuzz.rs]: I just haven't done fuzz yet, but probably.

File: README.md (hunk 3 above)
Comment (logannc) [reply README.md]: It's based on the python version, not the rust version.

**Comment by logannc** [comment]
In the future, can you make a new branch + PR back instead of pushing to one of my branches? All of the changes you made were fine, but I prefer my branches to not change out from underneath me ;)

This is primarily aimed at branches that touch code rather than docs, I just wanted to mention it now before it could become an issue.

**Comment by seanpianka** [comment]
No worries and will do. I've done this in other projects where multiple contributors to a single branch per feature/ticket isn't unexpected, so that's where the behavior comes from :)

**Review by seanpianka — APPROVED** [review]
